sdk/resourcemanager/azure-resourcemanager-keyvault/src/test/java/com/azure/resourcemanager/keyvault/ManagedHsmTests.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.keyvault;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.Region;
+import com.azure.core.util.Context;
+import com.azure.resourcemanager.keyvault.fluent.models.ManagedHsmInner;
+import com.azure.resourcemanager.keyvault.models.ManagedHsm;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmProperties;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmSku;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmSkuFamily;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmSkuName;
+import com.azure.resourcemanager.keyvault.models.MhsmNetworkRuleSet;
+import com.azure.resourcemanager.keyvault.models.PublicNetworkAccess;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ *
+ */
+public class ManagedHsmTests extends KeyVaultManagementTest {
+    @Override
+    protected void cleanUpResources() {
+        if (rgName != null) {
+            keyVaultManager.resourceManager().resourceGroups().beginDeleteByName(rgName);
+        }
+    }
+
+    @BeforeAll
+    // TODO(xiaofei) remove once core-management is released
+    public static void setup() {
+        AzureEnvironment.AZURE.getEndpoints().put("managedHsm", ".managedhsm.azure.net");
+    }
+
+    @Test
+    public void canCrudManagedHsms() {
+        String mhsmName = generateRandomResourceName("mhsm", 10);
```

Comment:
we probably want warning on both tests, stating that HSM must be deleted and purged ASAP (one could stop the test and not gets to finally).

Reply:
Sure, indeed.